src/Integration.Vsix/SolutionAnalysisRequester.cs

```diff
@@ -0,0 +1,141 @@
+                Debug.Fail($"{optionServiceTypeName} could not be found in {CodeAnalysisWorkspacesAssemblyName}");
+                return;
+            }
+
+            this.optionService = this.workspace.Services.GetType()
+                ?.GetMethod("GetService")
+                ?.MakeGenericMethod(optionServiceType)
+                ?.Invoke(workspace.Services, null);
+
+            Debug.Assert(this.optionService != null, "Option service is null");
+
+            this.optionServiceSetOptionsMethod = optionServiceType?.GetMethod("SetOptions");
+            this.optionServiceGetOptionsMethod = optionServiceType?.GetMethod("GetOptions");
+
+            Debug.Assert(this.optionServiceSetOptionsMethod != null, "IOptionService.SetOptions method is not found");
+            Debug.Assert(this.optionServiceGetOptionsMethod != null, "IOptionService.GetOptions method is not found");
+        }
+
+        private void FindFullSolutionAnalysisOption()
+        {
+            string codeAnalysisDllPath = typeof(Accessibility).Assembly.Location;
+            if (string.IsNullOrEmpty(codeAnalysisDllPath))
+            {
+                Debug.Fail($"Microsoft.CodeAnalysis.dll could not be located");
+                return;
+            }
+
+            FileInfo codeAnalysisAssembly = new FileInfo(codeAnalysisDllPath);
+            if (!codeAnalysisAssembly.Exists)
+            {
+                Debug.Fail($"Microsoft.CodeAnalysis.dll could not be located");
+                return;
+            }
+
+            string codeAnalysisFolderPath = codeAnalysisAssembly.Directory.FullName;
+            const string codeAnalysisFeaturesDllName = "Microsoft.CodeAnalysis.Features.dll";
+
+            //there's no public type in the DLL, so we try finding it next to Microsoft.CodeAnalysis
+            string path = Path.Combine(codeAnalysisFolderPath, codeAnalysisFeaturesDllName);
+
+            if (!File.Exists(path))
+            {
+                VsShellUtils.WriteToGeneralOutputPane(this.serviceProvider, Strings.MissingResourceAtLocation,
+                    codeAnalysisFeaturesDllName, codeAnalysisFolderPath);
+                Debug.Fail($"{codeAnalysisFeaturesDllName} could not be found in {codeAnalysisFolderPath}");
+                return;
+            }
+
+            this.option = (Option<bool>)Assembly.LoadFile(path)
+                .GetType("Microsoft.CodeAnalysis.Shared.Options.RuntimeOptions", false)
+                ?.GetField("FullSolutionAnalysis")
+                ?.GetValue(null);
+
+            Debug.Assert(this.option != null, "RuntimeOptions is not found");
+        }
+
+        private void FlipFullSolutionAnalysisFlag()
+        {
+            if (this.optionService == null ||
+                this.option == null)
+            {
+                return;
+            }
+
+            var options = this.optionServiceGetOptionsMethod?.Invoke(optionService, null) as OptionSet;
```

Comment:
You can check if null along with the other null checks and get rid of ? (also applies to another case a few lines below).